src/qibolab/instruments/zhinst.py

```diff
@@ -0,0 +1,1050 @@
+
+        # FIXME: Include this on the reports
+        # html containing the pulse sequence schedule
+        # lo.show_pulse_sheet("pulses", self.exp)
+
+        # There is no reason for disconnection and it prevents reconnection
+        # for a period of time making the software loops with execute_play_sequence crash
+        # self.disconnect()
+        return results
+
+    def sweep_recursion(self, qubits, exp, exp_calib, exp_options):
+        """Sweepers recursion for multiple nested Real Time sweepers"""
+
+        # This would reorder sweepers without the user knowing. I would like to avoid
+        # it as we should keep qibocal ordering.
+        # for sweep in self.sweepers:
+        #     if sweep.parameter is Parameter.frequency:
+        #         sweeper = sweep
+        #         break
+        #     sweeper = sweep
+
+        sweeper = self.sweepers[0]
+
+        i = len(self.sweepers) - 1
+        self.sweepers.remove(sweeper)
+        parameter = None
+
+        if sweeper.parameter is Parameter.frequency:
+            for pulse in sweeper.pulses:
+                line = "drive" if pulse.type is PulseType.DRIVE else "measure"
+                zhsweeper = ZhSweeper(pulse, sweeper, qubits[sweeper.pulses[0].qubit]).zhsweeper
+                zhsweeper.uid = f"frequency"  # TODO: Changing the name from "frequency" breaks it
+                exp_calib[f"{line}{pulse.qubit}"] = lo.SignalCalibration(
+                    oscillator=lo.Oscillator(
+                        frequency=zhsweeper,
+                        modulation_type=lo.ModulationType.HARDWARE,
+                    )
+                )
+        if sweeper.parameter is Parameter.amplitude:
+            for pulse in sweeper.pulses:
+                pulse.amplitude = pulse.amplitude * max(sweeper.values)
+                sweeper.values = sweeper.values / max(sweeper.values)
+                parameter = ZhSweeper(pulse, sweeper, qubits[sweeper.pulses[0].qubit]).zhsweeper
+
+        if sweeper.parameter is Parameter.bias:
+            for qubit in sweeper.qubits:
+                parameter = ZhSweeperLine(sweeper, qubit, self.sequence_qibo).zhsweeper
+
+        elif sweeper.parameter is Parameter.delay:
+            parameter = ZhSweeperLine(sweeper).zhsweeper
+
+        elif parameter is None:
+            parameter = ZhSweeper(sweeper.pulses[0], sweeper, qubits[sweeper.pulses[0].qubit]).zhsweeper
+
+        with exp.sweep(
+            uid=f"sweep_{sweeper.parameter.name.lower()}_{i}",  # FIXME: This uid trouble double freq ???
+            parameter=parameter,  # FIXME: This uid for double freq ???
+            reset_oscillator_phase=True,  # FIXME: Should we reset this phase ???
+        ):
+            if len(self.sweepers) > 0:
+                self.sweep_recursion(qubits, exp, exp_calib, exp_options)
+            else:
+                self.select_exp(exp, qubits, exp_options)
+
+    def sweep_recursion_NT(self, qubits, options, exp, exp_calib):
```

Comment:
Consider using customary case convention
```suggestion
    def sweep_recursion_nt(self, qubits, options, exp, exp_calib):
```